main.go

```diff
@@ -735,11 +748,9 @@ func main() {
 			const delay = 2 * time.Second
 			for {
 				deps.l.Warnf("Retrying in %s.", delay)
-				sleepCtx, sleepCancel := context.WithTimeout(ctx, delay)
-				<-sleepCtx.Done()
-				sleepCancel()
-
-				if ctx.Err() != nil {
+				select {
+				case <-time.After(delay):
+				case <-ctx.Done():
```

Comment:
why did you change this behaviour? it ignores ctx.Done()

Reply:
ok, returned back